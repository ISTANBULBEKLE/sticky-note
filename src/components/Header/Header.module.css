.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  background: linear-gradient(135deg, #6366f1, #8b5cf6);
  color: white;
}

.logo h1 {
  font-size: 1.5rem;
  font-weight: bold;
  margin: 0;
}

.nav {
  display: flex;
  gap: 2rem;
}

.link {
  color: white;
  text-decoration: none;
  font-weight: 500;
  transition: opacity 0.2s;
}

.link:hover {
  opacity: 0.8;
} 