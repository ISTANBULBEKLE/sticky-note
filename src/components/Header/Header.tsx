import Link from 'next/link';
import styles from './Header.module.css';

const Header = () => {
  return (
    <header className={styles.header}>
      <div className={styles.logo}>
        <h1>Sticky Notes</h1>
      </div>
      <nav className={styles.nav}>
        <Link href="/" className={styles.link}>Home</Link>
        <Link href="/contact" className={styles.link}>Contact</Link>
      </nav>
    </header>
  );
};

export default Header; 