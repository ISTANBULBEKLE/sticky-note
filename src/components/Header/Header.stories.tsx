import type { Meta, StoryObj } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Header from './Header';

const meta = {
  title: 'Components/Header',
  component: Header,
  parameters: {
    layout: 'fullscreen',
    docs: {
      description: {
        component: 'Main navigation header component with logo and navigation links.',
      },
    },
  },
  tags: ['autodocs'],
  // Mock Next.js Link component for Storybook
  decorators: [
    (Story) => (
      <div style={{ margin: '-1rem' }}>
        <Story />
      </div>
    ),
  ],
} satisfies Meta<typeof Header>;

export default meta;
type Story = StoryObj<typeof Header>;

// Default story
export const Default: Story = {
  args: {},
};

// Story demonstrating home navigation
export const NavigatingHome: Story = {
  parameters: {
    docs: {
      description: {
        story: 'Clicking the Home link triggers navigation.',
      },
    },
  },
  play: async ({ canvasElement }) => {
    const homeLink = canvasElement.querySelector('a[href="/"]');
    homeLink?.addEventListener('click', (e) => {
      e.preventDefault();
      action('Navigate to Home')();
    });
  },
};

// Story demonstrating contact navigation
export const NavigatingToContact: Story = {
  parameters: {
    docs: {
      description: {
        story: 'Clicking the Contact link triggers navigation.',
      },
    },
  },
  play: async ({ canvasElement }) => {
    const contactLink = canvasElement.querySelector('a[href="/contact"]');
    contactLink?.addEventListener('click', (e) => {
      e.preventDefault();
      action('Navigate to Contact')();
    });
  },
};

// Story with dark theme
export const DarkTheme: Story = {
  parameters: {
    backgrounds: {
      default: 'dark',
    },
  },
  decorators: [
    (Story) => (
      <div style={{ margin: '-1rem', background: '#1a1a1a' }}>
        <Story />
      </div>
    ),
  ],
};