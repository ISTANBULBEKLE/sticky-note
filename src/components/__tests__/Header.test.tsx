import { render, screen } from '@testing-library/react';
import Header from '../Header/Header';

// Mock the next/link component
jest.mock('next/link', () => {
  return ({ children, href }: { children: React.ReactNode; href: string }) => {
    return <a href={href}>{children}</a>;
  };
});

describe('Header', () => {
  it('renders the header with title and navigation links', () => {
    render(<Header />);
    
    // Check if the title is rendered
    expect(screen.getByText('Sticky Notes')).toBeInTheDocument();
    
    // Check if navigation links are rendered
    const homeLink = screen.getByRole('link', { name: /home/i });
    const contactLink = screen.getByRole('link', { name: /contact/i });
    
    expect(homeLink).toBeInTheDocument();
    expect(homeLink).toHaveAttribute('href', '/');
    
    expect(contactLink).toBeInTheDocument();
    expect(contactLink).toHaveAttribute('href', '/contact');
  });

  it('renders with proper structure and styling', () => {
    const { container } = render(<Header />);
    
    // Check if header exists
    const header = container.querySelector('header');
    expect(header).toBeInTheDocument();
    
    // Check if logo div exists
    const logo = container.querySelector('div');
    expect(logo).toBeInTheDocument();
    
    // Check if nav exists
    const nav = container.querySelector('nav');
    expect(nav).toBeInTheDocument();
    
    // Check if links are inside nav
    const links = nav?.querySelectorAll('a');
    expect(links?.length).toBe(2);
  });

  it('matches snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
}); 