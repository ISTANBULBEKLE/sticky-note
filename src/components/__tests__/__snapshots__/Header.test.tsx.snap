// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="logo"
    >
      <h1>
        Sticky Notes
      </h1>
    </div>
    <nav
      class="nav"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/contact"
      >
        Contact
      </a>
    </nav>
  </header>
</div>
`;
